src/matrix_multiply_add.jl

```diff
@@ -0,0 +1,352 @@
+        a::StaticMatrix, b::StaticMatrix, _add::MulAddMul) where {sa, sb, sc, tA, tB}
+    if sb[1] != sa[2] || sa[1] != sc[1] || sb[2] != sc[2]
+        throw(DimensionMismatch("Tried to multiply arrays of size $sa and $sb and assign to array of size $sc"))
+    end
+
+    #vect_exprs = [:($(Symbol("tmp_$k2")) = partly_unrolled_multiply(A, B[:, $k2])) for k2 = 1:sB[2]]
+
+    # Do a custom b[:, k2] to return a SVector (an isbitstype type) rather than a mutable type. Avoids allocation == faster
+    tmp_type = SVector{sb[1], eltype(c)}
+    vect_exprs = [:($(Symbol("tmp_$k2")) = partly_unrolled_multiply($(TSize{sa,tA}()), $(TSize{(sb[1],),tB}()),
+        a, $(Expr(:call, tmp_type, [:($(_lind(:b, Sb, i, k2))) for i = 1:sb[1]]...)))) for k2 = 1:sb[2]]
+
+    lhs = [:($(_lind(:c, Sc, k1, k2))) for k1 = 1:sa[1], k2 = 1:sb[2]]
+    # exprs = [:(c[$(LinearIndices(sc)[k1, k2])] = $(Symbol("tmp_$k2"))[$k1]) for k1 = 1:sa[1], k2 = 1:sb[2]]
+    rhs = [:($(Symbol("tmp_$k2"))[$k1]) for k1 = 1:sa[1], k2 = 1:sb[2]]
+    exprs = _muladd_expr(lhs, rhs, _add)
+
+    return quote
+        @_inline_meta
+        # α = _add.alpha
+        # β = _add.beta
+        α = alpha(_add)
+        β = beta(_add)
+        @inbounds $(Expr(:block, vect_exprs...))
+        @inbounds $(Expr(:block, exprs...))
+    end
+end
+
+# @inline partly_unrolled_multiply(Sa::Size, Sb::Size, a::StaticMatrix, b::StaticArray) where {sa, sb, Ta, Tb} =
+#     partly_unrolled_multiply(TSize(Sa), TSize(Sb), a, b)
+@generated function partly_unrolled_multiply(Sa::TSize{sa}, ::TSize{sb}, a::StaticMatrix{<:Any, <:Any, Ta}, b::StaticArray{<:Tuple, Tb}) where {sa, sb, Ta, Tb}
+    if sa[2] != sb[1]
+        throw(DimensionMismatch("Tried to multiply arrays of size $sa and $sb"))
+    end
+
+    if sa[2] != 0
+        exprs = [reduce((ex1,ex2) -> :(+($ex1,$ex2)), [:($(_lind(:a,Sa,k,j))*b[$j]) for j = 1:sa[2]]) for k = 1:sa[1]]
+    else
+        exprs = [:(zero(promote_op(matprod,Ta,Tb))) for k = 1:sa[1]]
+    end
+
+    return quote
+        $(Expr(:meta,:noinline))
+        @inbounds return SVector(tuple($(exprs...)))
+    end
+end
+
+@inline _get_raw_data(A::SizedArray) = A.data
+@inline _get_raw_data(A::StaticArray) = A
+
+function mul_blas!(::TSize{<:Any,false}, c::StaticMatrix, ::TSize{<:Any,tA}, ::TSize{<:Any,tB},
+        a::StaticMatrix, b::StaticMatrix, _add::MulAddMul) where {tA,tB}
+    mat_char(tA) = tA ? 'T' : 'N'
+    T = eltype(a)
+    A = _get_raw_data(a)
+    B = _get_raw_data(b)
+    C = _get_raw_data(c)
+    BLAS.gemm!(mat_char(tA), mat_char(tB), T(alpha(_add)), A, B, T(beta(_add)), C)
+end
+
+# if C is transposed, transpose the entire expression
+@inline mul_blas!(Sc::TSize{<:Any,true}, c::StaticMatrix, Sa::TSize, Sb::TSize,
+        a::StaticMatrix, b::StaticMatrix, _add::MulAddMul) =
+    mul_blas!(transpose(Sc), c, transpose(Sb), transpose(Sa), b, a, _add)
+
+# TODO: Get this version of mul_blas! working so there's backward-compatibility with older Julia versions
```

Comment:
Is this TODO still needed?

Reply:
Nope. Fixed and deleted out the old code.